### PR TITLE
Add Link Checker Report Github Action

### LINKED ISSUE #85: Create a link checker CI
Currently the tour has many links in the [/content](https://github.com/json-schema-org/tour/tree/main/content), Create a CI that is similar to what we have in the website repo ([workflow file](https://github.com/json-schema-org/website/blob/main/.github/workflows/link-checker.yml)), that creates an issue similar to [this one](https://github.com/json-schema-org/website/issues/986) with list of all the errored links in the project.

### PR DESCRIPTION
**Pull request by heysujal**
**What kind of change does this PR introduce?**

**Issue Number:**

-  Closes #85  

**Screenshots/videos:**

First, I added a broken link 
image

[This issue ](https://github.com/heysujal/tour/issues/4) got created 
image

**If relevant, did you update the documentation?**

**Summary**
This is a github action which creates an issue reporting all the bad/broken links that has been found in all the `.mdx` files inside `content/`

**Does this PR introduce a breaking change?**
No

### CODE REVIEW

File: .github/workflows/link-checker.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          path: .lycheecache
+          key: cache-lychee-${{ github.sha }}
+          restore-keys: cache-lychee-
+
+      # This will run the link checker on all markdown files in the pages directory      
+      - name: Link Checker
+        id: lychee
+        uses: lycheeverse/lychee-action@v1
+        with:
+          args: --base https://json-schema.org  --verbose --no-progress --accept 200,204,429,403  './content/**/*.mdx' --cache --max-cache-age 1d  https://json-schema.org
```
Comment (JeelRajodiya) [inline .github/workflows/link-checker.yml:56]: I am not sure how the link checker workflow works, but why do we have json-shcema.org passed in the arguments? shouldn't it be tour.json-schema.org?  and if the workflow checks in the ./content/**/*.mdx files (which is what we want) then why are we passing the url of the website here?

**Review by JeelRajodiya — CHANGES_REQUESTED** [review]

**Comment by JeelRajodiya** [comment]
@heysujal I saw the [issue created](https://github.com/heysujal/tour/issues/4) by the workflow. 
in the issue, why the link is just somebadlink.com?  shouldn't it be an actual bad link? 
image

File: .github/workflows/link-checker.yml (hunk 1 above)
Comment (heysujal) [reply .github/workflows/link-checker.yml:56]: I see. I will have a look at it again.
I used the mentioned example workflow code to work upon. I will see what it is being used for and if we can remove it totally.

Comment (heysujal) [reply .github/workflows/link-checker.yml:56]: I found that the link at the starting with `--base` is required if we are using relative links in markdown. 
The link at the end is added to make check it too in addition to all the links in `.mdx`. I think this is being used to check if our site itself is down which might be lead to links in mardown files to fail.
I will update the url with `https://tour.json-schema.org`

Comment (JeelRajodiya) [reply .github/workflows/link-checker.yml]: Thanks @heysujal.

**Pull request merged**